components/dashboard/src/Insights.tsx

```diff
@@ -0,0 +1,214 @@
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@podkit/select/Select";
+import dayjs from "dayjs";
+import { Timestamp } from "@bufbuild/protobuf";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+import { TextMuted } from "@podkit/typography/TextMuted";
+import { DownloadInsightsToast } from "./insights/download/DownloadInsights";
+import { useCurrentOrg } from "./data/organizations/orgs-query";
+import { useToast } from "./components/toasts/Toasts";
+import { useTemporaryState } from "./hooks/use-temporary-value";
+import { DownloadIcon } from "lucide-react";
+import { Button } from "@podkit/buttons/Button";
+
+export const Insights = () => {
+    const [prebuildsFilter, setPrebuildsFilter] = useState<"week" | "month" | "year">("week");
+    const [upperBound, lowerBound] = useMemo(() => {
+        const from = dayjs().subtract(1, prebuildsFilter);
+
+        const fromTimestamp = Timestamp.fromDate(from.toDate());
+        const toTimestamp = Timestamp.fromDate(new Date());
+        return [fromTimestamp, toTimestamp];
+    }, [prebuildsFilter]);
+    const {
+        data,
+        error: errorMessage,
+        isLoading,
+        isFetchingNextPage,
+        hasNextPage,
+        fetchNextPage,
+    } = useWorkspaceSessions({
+        from: upperBound,
+        to: lowerBound,
+    });
+    const membersQuery = useListOrganizationMembers();
+    const members: OrganizationMember[] = useMemo(() => membersQuery.data || [], [membersQuery.data]);
+
+    const hasMoreThanOnePage = (data?.pages.length ?? 0) > 1;
+    const sessions = useMemo(() => data?.pages.flatMap((p) => p) ?? [], [data]);
+    const grouped = Object.groupBy(sessions, (ws) => ws.workspace?.id ?? "unknown");
+    const [page, setPage] = useState(0);
+
+    return (
+        <>
+            <Header title="Insights" subtitle="Insights into workspace sessions in your organization" />
+            <div className="app-container pt-5">
+                <div
+                    className={classNames(
+                        "flex flex-col items-start space-y-3 justify-between",
+                        "md:flex-row md:items-center md:space-x-4 md:space-y-0",
+                    )}
+                >
+                    <Select value={prebuildsFilter} onValueChange={(v) => setPrebuildsFilter(v as any)}>
+                        <SelectTrigger className="w-[180px]">
+                            <SelectValue placeholder="Select time range" />
+                        </SelectTrigger>
+                        <SelectContent>
+                            <SelectItem value="day">Last 24 hours</SelectItem>{" "}
+                            {/* here for debugging, probably not useful */}
+                            <SelectItem value="week">Last 7 days</SelectItem>
+                            <SelectItem value="month">Last 30 days</SelectItem>
+                            <SelectItem value="year">Last 365 days</SelectItem>
```

Comment:
I did not see a good reason for being able to select an absolute time range (like we use today with the usage dashboard), so I just made it use some relative options.